### ISSUE TITLE
sd_journal does not handle journal moving to persistent logging

### ISSUE THREAD

**Issue by lnykryn** [systemd/systemd#8262]
### Submission type

 - Bug report

### systemd version the issue has been seen with

> latest upstream, but it never worked

### In case of bug report: Expected behaviour you didn't see

> When we have a runtime journal, and we create /var/log/journal/, journald will start to log there, but anything that uses sd_journal api will not get the new logs.

### In case of bug report: Steps to reproduce the problem

> journalctl -f
> mkdir /var/log/journal/ ; journalctl --flush ; logger foo

**Comment by poettering** [comment]
yeah, we only watch the logging "root" directories that exist when journalctl is started... Watching the creation of those directories is nasty but doable (.path units do this after all...)
